Homework/2/Program.cs

```diff
@@ -0,0 +1,390 @@
+        static void ListUsers() {
+            Console.Clear();
+            Console.WriteLine("Users:\n");
+
+            for (int i = 0; i < userArray.Length; i++)
+            {
+                Console.WriteLine(userArray[i][0]);
+            }
+            PromptInput("\nPress any key to continue.");
+            MenuSelection();
+        }
+
+        //Function to add userDetails to end or start of userArray. Taking bool variables to determine whether add to start or end or select index.
+        static void PromptAddUserToArray(bool addToEnd = true,bool addToStart = false) {
+            Console.Clear();
+            Console.WriteLine("Adding user:\n");
+
+            string username = PromptInput("Enter username:");
+            string password = PromptInput("Enter password:");
+            
+            int index = userArray.Length;
+
+            if (addToStart)
+            {
+                index = 0;
+            }
+
+            if (!addToEnd && !addToStart)
+            {
+                string indexStr = PromptInput("Enter index to add user at:");
+                index = int.Parse(indexStr);
+            }
+
+            if (index<0 || index > userArray.Length)
+            {
+                if (RetryConfirmation("Bad index."))
+                {
+                    PromptAddUserToArray(addToEnd,addToStart);
+                }
+                else
+                {
+                    MenuSelection();
+                }
+            }
+
+            if (!CheckIfUserExists(username) || !(password == ""))
+            {
+                string[] userDetails = new string[2] { username, password };
+                AddUserToArrayAtIndex(userDetails, index);
+                ListUsers();
+                return;
+            }
+            else {
+                if (RetryConfirmation("Bad user details."))
+                {
+                    PromptAddUserToArray(addToEnd, addToStart);
+                }
+                else
+                {
+                    MenuSelection();
+                }
+            }
+        }
+
+        //Function to get index of where user should be removed
+        static void PromptRemoveUser(bool removeFromEnd = true, bool removeFromStart = false) {
+            Console.Clear();
+
+            int index = userArray.Length - 1;
+
+            if (removeFromStart)
+            {
+                index = 0;
+            }
+
+            if (!removeFromEnd && !removeFromStart)
+            {
+                string indexStr = PromptInput("Enter index to remove user at:");
+                index = int.Parse(indexStr);
+            }
+
+            if (index < 0 || index >= userArray.Length)
+            {
+                if (RetryConfirmation("Bad index."))
+                {
+                    PromptRemoveUser(removeFromEnd, removeFromStart);
+                }
+                else
+                {
+                    MenuSelection();
+                }
+            }
```

Comment:
This block of code should be in a new function called `PromptIndex`. That way, you would have a high level function which: gets index, removes user at that index and refreshes the list on screen. This makes readability of this function a ton better, doesn't it? Right now, with the current situation, you mix the above code with high level calls like `RemoveUser` and `ListUsers` which does makes this function do more than one thing- which it shouldn't.